BoughtTogether/BoughtTogether.java

```diff
@@ -0,0 +1,93 @@
+import java.io.IOException;
+
+import java.util.regex.Pattern;
+import java.util.Arrays;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.io.IntWritable;
+import org.apache.hadoop.io.Text;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.mapreduce.Job;
+import org.apache.hadoop.mapreduce.Mapper;
+import org.apache.hadoop.mapreduce.Reducer;
+import org.apache.hadoop.mapreduce.lib.input.TextInputFormat;
+import org.apache.hadoop.mapreduce.lib.output.FileOutputFormat;
+
+public class BoughtTogether {
+
+    /* generate combinations of items that were bought together */
+    public static class LineMapper
+        extends Mapper<Object, Text, Text, IntWritable> {
+
+        private final static Pattern itemSeparator = Pattern.compile(", *");
+        private final static IntWritable one = new IntWritable(1);
+
+        public void map(Object key, Text value, Context context)
+            throws IOException, InterruptedException {
+
+            /* input lines */
+            String line = value.toString();
+
+            /* split comma-separated items */
+            String[] items = itemSeparator.split(line);
+
+            /* sort the items to prevent duplicates */
+            Arrays.sort(items);
```

Comment:
thanks for mentioning this, i should update the comment. the reason i am sorting is so that the output pairs "Bananas, Best Bread" and "Best Bread, Bananas" aren't duplicated. duplicates should be counted correctly in the mapper.